src/integrationTest/java/org/opensearch/security/rest/AuthZinRestLayerTests.java

```diff
@@ -0,0 +1,254 @@
+        new Role("dummy_rest_only_role").clusterPermissions("security:dummy_protected/get")
+            .clusterPermissions("cluster:admin/dummy_plugin/dummy")
+    );
+
+    protected final static TestSecurityConfig.User DUMMY_WITH_TRANSPORT_PERM = new TestSecurityConfig.User("dummy_transport_perm").roles(
+        new Role("dummy_transport_perm_role").clusterPermissions("security:dummy_protected/get")
+            .clusterPermissions("cluster:admin/dummy_plugin/dummy", "cluster:admin/dummy_protected_plugin/dummy/get")
+    );
+
+    protected final static TestSecurityConfig.User DUMMY_LEGACY = new TestSecurityConfig.User("dummy_user_legacy").roles(
+        new Role("dummy_role_legacy").clusterPermissions("cluster:admin/dummy_plugin/dummy")
+    );
+
+    protected final static TestSecurityConfig.User DUMMY_NO_PERM = new TestSecurityConfig.User("dummy_user_no_perm").roles(
+        new Role("dummy_role_no_perm")
+    );
+
+    protected final static TestSecurityConfig.User DUMMY_UNREGISTERED = new TestSecurityConfig.User("dummy_user_not_registered");
+
+    public static final String DUMMY_BASE_ENDPOINT = "_plugins/_dummy";
+    public static final String DUMMY_PROTECTED_BASE_ENDPOINT = "_plugins/_dummy_protected";
+    public static final String DUMMY_API = DUMMY_BASE_ENDPOINT + "/dummy";
+    public static final String DUMMY_PROTECTED_API = DUMMY_PROTECTED_BASE_ENDPOINT + "/dummy";
+
+    @ClassRule
+    public static LocalCluster cluster = new LocalCluster.Builder().clusterManager(ClusterManager.THREE_CLUSTER_MANAGERS)
+        .authc(AUTHC_HTTPBASIC_INTERNAL)
+        .users(DUMMY_REST_ONLY, DUMMY_WITH_TRANSPORT_PERM, DUMMY_LEGACY, DUMMY_NO_PERM)
+        .plugin(CustomLegacyTestPlugin.class)
+        .plugin(CustomRestProtectedTestPlugin.class)
+        .audit(
+            new AuditConfiguration(true).compliance(new AuditCompliance().enabled(true))
+                .filters(new AuditFilters().enabledRest(true).enabledTransport(true).resolveBulkRequests(true))
+        )
+        .build();
+
+    @Rule
+    public AuditLogsRule auditLogsRule = new AuditLogsRule();
+
+    /** Basic Access check */
+
+    @Test
+    public void testShouldFailForUnregisteredUsers() {
+        try (TestRestClient client = cluster.getRestClient(DUMMY_UNREGISTERED)) {
+            // Legacy plugin
+            assertThat(client.get(DUMMY_API).getStatusCode(), equalTo(HttpStatus.SC_UNAUTHORIZED));
+            auditLogsRule.assertExactlyOne(privilegePredicateRESTLayer(FAILED_LOGIN, DUMMY_UNREGISTERED, GET, "/" + DUMMY_API));
+
+            // Protected Routes plugin
+            assertThat(client.get(DUMMY_PROTECTED_API).getStatusCode(), equalTo(HttpStatus.SC_UNAUTHORIZED));
+            auditLogsRule.assertExactlyOne(privilegePredicateRESTLayer(FAILED_LOGIN, DUMMY_UNREGISTERED, GET, "/" + DUMMY_PROTECTED_API));
+        }
+    }
+
+    @Test
+    public void testShouldFailForBothPlugins() {
```

Comment:
DUMMY_API is not a NamedRoute so there is no REST layer authz check, and so the check will directly move to Transport layer where it will fail due to lack of permissions for DUMMY_NO_PERM user

For second case, DUMMY_PROTECTED_API is a NamedRoute, so the check would fail at REST layer since the user doesn't have permission to this NamedRoute